### PR TITLE
Adding migration for stock_exchange

### PR DESCRIPTION
**Pull request by RohanDudeja**

### CODE REVIEW

**Comment by mukesh0513** [comment]
remove last file.

**Comment by RohanDudeja** [comment]
@mukesh0513 removed stock_exchange.db

**Comment by mukesh0513** [comment]
why have we created a stock exchange folder under migration? it will increase the complexity of running migrations available in different folders.

File: migration/stock_exchange/20210803123705_stocks.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    open_price int,
+    stock_ticker_symbol varchar,
+    ltp int,
+    high_price int,
+    low_price int,
+    previous_day_close int,
+    percentage_change int,
+    created_at timestamp,
+    updated_at timestamp,
+    deleted_at timestamp
```
Comment (mukesh0513) [inline migration/stock_exchange/20210803123705_stocks.sql]: add primary key.

File: migration/stock_exchange/20210803123705_stocks.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS STOCKS(
+    id int not null,
```
Comment (mukesh0513) [inline migration/stock_exchange/20210803123705_stocks.sql]: how would unique id be generated?

File: migration/stock_exchange/20210803130807_buy_order_book.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS BUY_ORDER_BOOK(
+    order_id varchar not null,
+    stock_ticker_symbol varchar,
+    order_quantity int,
+    order_status varchar,
+    order_price varchar,
+    created_at timestamp,
+    updated_at timestamp,
+    deleted_at timestamp
```
Comment (mukesh0513) [inline migration/stock_exchange/20210803130807_buy_order_book.sql]: primary key?

Comment (RohanDudeja) [reply migration/stock_exchange/20210803130807_buy_order_book.sql]: added primary key

File: migration/stock_exchange/20210803123705_stocks.sql (hunk 2 above)
Comment (RohanDudeja) [reply migration/stock_exchange/20210803123705_stocks.sql]: added auto_increment

File: migration/20210803123705_stocks.sql [hunk 4]
```diff
@@ -0,0 +1,22 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS stocks(
+    id int not null unique AUTO_INCREMENT,
```
Comment (mukesh0513) [inline migration/20210803123705_stocks.sql]: there can be multiple unique key in a table. mention explicitly about primary key.

File: migration/20210803131854_sell_order_book.sql [hunk 5]
```diff
@@ -0,0 +1,18 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS sell_order_book(
```
Comment (mukesh0513) [inline migration/20210803131854_sell_order_book.sql:3]: add default values wherever needed.

File: migration/20210803123705_stocks.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS stocks(
+    id int not null unique AUTO_INCREMENT primary key,
```
Comment (mukesh0513) [inline migration/20210803123705_stocks.sql]: primary key is already unique.

File: migration/20210803123705_stocks.sql [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    open_price int not null,
+    stock_ticker_symbol varchar(255) not null,
+    ltp int not null,
+    high_price int not null,
+    low_price int not null,
+    previous_day_close int not null,
+    percentage_change int not null,
+    created_at timestamp default current_timestamp,
+    updated_at timestamp default current_timestamp,
+    deleted_at timestamp
```
Comment (mukesh0513) [inline migration/20210803123705_stocks.sql]: shouldn't this be default null?

File: migration/20210803123705_stocks.sql [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    id int not null AUTO_INCREMENT primary key,
+    stock_name varchar(255) not null,
+    open_price int not null,
+    stock_ticker_symbol varchar(255) not null,
+    ltp int not null,
+    high_price int not null,
+    low_price int not null,
+    previous_day_close int not null,
+    percentage_change int not null,
+    created_at timestamp default current_timestamp,
```
Comment (mukesh0513) [inline migration/20210803123705_stocks.sql]: adding not null will suffice.

File: migration/20210803130807_buy_order_book.sql [hunk 9]
```diff
@@ -0,0 +1,19 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE IF NOT EXISTS buy_order_book(
+    id int not null AUTO_INCREMENT primary key,
+    order_id varchar(255) not null,
+    stock_ticker_symbol varchar(255) not null,
+    order_quantity int not null,
+    order_status varchar(255) default 'pending',
```
Comment (mukesh0513) [inline migration/20210803130807_buy_order_book.sql]: Same comment as above. Check all migration files.

**Review by mukesh0513 — APPROVED** [review]
LGTM

**Pull request merged**